### ISSUE TITLE
Customizable return type, parameters and attributes

### ISSUE THREAD

**Issue by persn** [zompinc/sync-method-generator#43]
Using `#if SYNC_ONLY` allows us to customize the code within the method block, however customizing the rest of the method is also useful.

Adding attributes like `[Obsolete]` for instance would be useful because we typically only want to have one method, but we still need to support the sync version for our API consumers until it can be removed altogether. Other attributes are probably also useful

```csharp
[Zomp.SyncMethodGenerator.CreateSyncVersion]
#if SYNC_ONLY
[Obsolete]
[ExcludeFromCodeCoverage]
#endif
public async Task FooAsync()
{
    await Task.Delay(1000);
}
```

Customizing parameters are useful because sometimes sometimes sync and async api exists on different types. For example `System.Data.IDataReader` has only sync methods, when Microsoft wanted to extend async methods they kept the interface as-is and made a new type `System.Data.Common.DbDataReader`. So the following code

```csharp
public bool IsNull(IDataReader reader, int i)
{
    return reader.IsDBNull(i);
}

public async Task<bool> IsNullAsync(DbDataReader reader, int i)
{
    return await reader.IsDBNullAsync(i);
}
```

Could maybe become

```csharp
[Zomp.SyncMethodGenerator.CreateSyncVersion]
public async Task<bool> IsNullAsync(
#if SYNC_ONLY
    IDataReader reader,
#else
    DbDataReader reader,
#endif
    int i)
{
    return await reader.IsDBNullAsync(i);
}
```

Customizing return types are useful for the same reason as parameter but unsure how that could look like.

**Comment by virzak** [comment]
Working on this in [attribute-trivia](https://github.com/zompinc/sync-method-generator/tree/attribute-trivia). This is not straightforward at all, especially inside parameters.

**Comment by virzak** [comment]
This is released in version 1.3.20

There might be test cases related to this functionality that aren't covered. If so, please open a new issue.
